File: qb-engineer-server/qb-engineer.core/Models/ScheduleMilestoneModel.cs
namespace QBEngineer.Core.Models;

public record ScheduleMilestoneModel(
    int SalesOrderLineId,
    string? PartNumber,
    string? PartDescription,
    DateTimeOffset? DeliveryDate,
    DateTimeOffset? ShipBy,
    DateTimeOffset? QcCompleteBy,
    DateTimeOffset? ProductionCompleteBy,
    DateTimeOffset? ProductionStartBy,
    DateTimeOffset? MaterialsNeededBy,
    DateTimeOffset? PoOrderBy,
    bool IsAtRisk);

File: qb-engineer-ui/src/app/features/sales-orders/models/schedule-milestone.model.ts
export interface ScheduleMilestone {
  salesOrderLineId: number;
  partNumber: string | null;
  partDescription: string | null;
  deliveryDate: string | null;
  shipBy: string | null;
  qcCompleteBy: string | null;
  productionCompleteBy: string | null;
  productionStartBy: string | null;
  materialsNeededBy: string | null;
  poOrderBy: string | null;
  isAtRisk: boolean;
}
